hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -513,19 +513,27 @@ private String intArrayToString(int[] ints) {
     @Override
     public <V> DataSet<LocatedRow> readTextFile(SpliceOperation op, String location, String characterDelimiter, String columnDelimiter, int[] baseColumnMap,
                                       OperationContext context, Qualifier[][] qualifiers, DataValueDescriptor probeValue, ExecRow execRow) throws StandardException {
+        assert baseColumnMap != null:"baseColumnMap Null";
         try {
             Dataset<Row> table = null;
             try {
                 table = SpliceSpark.getSession().read().csv(location);
 
-
-                for( int index = 0; index< baseColumnMap.length; index++){
-                    if(baseColumnMap[index]!=-1){
+                if (op == null && baseColumnMap[0] == 1) {
```

Comment:
My understanding is that baseColumnMap parameter tracks the columns that need to collect stats. So baseColumnMap[0] may not always be 1.  Is the check "baseColumnMap[0]==1" appropriate? If collect stats is disabled for the first column, will your fix still be triggered?